opensearch-fetcher/log_fetching_service.py

```diff
@@ -0,0 +1,122 @@
+# Standard Library
+import asyncio
+import logging
+import os
+import time
+
+# Third Party
+from datetime import datetime
+from elasticsearch import AsyncElasticsearch, TransportError
+from opni_nats import NatsWrapper
+import pandas as pd
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(name)s - %(message)s")
+
+ES_ENDPOINT = os.environ["ES_ENDPOINT"]
+ES_USERNAME = os.environ["ES_USERNAME"]
+ES_PASSWORD = os.environ["ES_PASSWORD"]
+TIME_RANGE_SECONDS = int(os.getenv("TIME_RANGE_SECONDS", "10"))
+
+nw = NatsWrapper()
+
+async def send_all_results_to_nats(es, start_ts, end_ts):
+    scroll_value = "1m"
+    es_query_body = {
+        "query": {
+            "bool": {
+                "filter": [{"range": {"time": {"gte": start_ts, "lt": end_ts}}}]
+            },
+        }
+    }
+
+    first_page = True
+    while True:
+        if first_page:
+            current_page = await es.search(index="logs", body=es_query_body, scroll=scroll_value, size=2000)
+            first_page = False
+        else:
+            current_page = await es.scroll(scroll_id=scroll_id, scroll=scroll_value)
+        if "_scroll_id" in current_page:
+            scroll_id = current_page["_scroll_id"]
+            current_page_results = current_page["hits"]["hits"]
+            if len(current_page_results) > 0:
+                accumulated_results = []
+                for result in current_page_results:
+                    result_dict = result["_source"].copy()
+                    result_dict["_id"] = result["_id"]
+                    accumulated_results.append(result_dict)
+                accumulated_results_df = pd.DataFrame(accumulated_results)
+                logging.info("Published {} logs to Nats now".format(len(accumulated_results)))
+                await nw.publish("raw_logs", accumulated_results_df.to_json().encode())
+            else:
+                break
+        else:
+            break
+
+async def fetch_logs():
+    es = await setup_es_connection()
+    try:
+        current_ts = int((datetime.now().timestamp() - TIME_RANGE_SECONDS) * 1000)
+        while True:
+            await asyncio.sleep(TIME_RANGE_SECONDS)
```

Comment:
I think the wait time should be small than `TIME_RANGE_SECONDS`

Reply:
The idea is every 10 seconds or so this function will run and fetch all the logs from the previous 10 seconds. The wait time is mainly put in so it fetches the logs at a fixed rate.

Reply:
have you tried to add a timer to track down the time here

Reply:
So I can add a timer but what if for some reason the query takes over 10 seconds to reach, then the wait time would be negative. Would need to add further logic then to handle that.